### PR TITLE
fix: ten edge-case correctness defects across assert, loader, and report

### PR DESCRIPTION
**Pull request by nao1215**
A single-session edge-case bug hunt over the assert, loader, and report surfaces. Ten distinct correctness defects, each reproduced with a test first and fixed in place; no new features and no behavior change on the common path. Full test suite, golangci-lint, and the 228-scenario self-hosted e2e all pass.

Numeric matchers (internal/assert). json/yaml equals rounded both operands through float64, so two distinct 64-bit ids (9007199254740993 vs 9007199254740992) compared equal; integers are now compared at full precision with math/big. gt/gte/lt/lte rejected any integer beyond int64 as "not numeric" because oj decodes it as a json.Number the numeric path ignored; json.Number is now accepted. matches rendered a whole-number float with %v, so 1000000.0 became "1e+06" and a digit pattern like ^1000000$ never matched; numeric nodes now render without scientific notation, and the equals failure message reads the same way.

Loader validation (internal/loader). An empty matches/not_matches pattern matches everything, so matches: "" is an always-true no-op and not_matches: "" can never pass; both are now rejected at load on the stream, json/yaml, and header matchers, mirroring the existing empty-string contains/not_contains rejection. A negative json/yaml length and a negative duration bound are likewise rejected, since no length is negative and no measured duration is below zero.

Report formats (internal/report). A control byte in captured output — most often an ANSI color escape — flowed verbatim into the machine-readable reports: in a TAP diagnostic it broke the YAML data: block ("control characters are not allowed"), and in a GitHub Actions annotation it reached the message unescaped. Both now fold such bytes to the Unicode replacement character, keeping tab and newline, as junit already does through its XML encoder; invalid UTF-8 is folded the same way so a TAP stream stays valid UTF-8.

https://claude.ai/code/session_01BtcgityD3NW5Q89fbbWH91

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved numeric comparisons so large integers and whole-number floats are handled correctly in assertions.
  * Tightened validation for test configurations by rejecting empty patterns and negative numeric bounds at load time.
  * Cleaned up failure output to avoid raw control characters, helping TAP, YAML, and GitHub Actions logs render safely and parse reliably.

### CODE REVIEW

File: internal/assert/jsonmatch.go [hunk 1]
```diff
@@ -268,6 +288,15 @@ func valuesEqual(node, want any) bool {
...
 	if isNumericKind(node) || isNumericKind(want) {
+		// Prefer exact integer comparison. Two distinct integers beyond 2^53 (a
+		// JSON id, or a value beyond int64 that decodes as json.Number) round to
+		// the same float64, so a float compare would report them equal. big.Int
+		// keeps every digit.
+		if ni, ok := toBigInt(node); ok {
+			if wi, ok := toBigInt(want); ok {
+				return ni.Cmp(wi) == 0
+			}
+		}
```
[the remark this answers is not in the document]
Comment (nao1215) [reply internal/assert/jsonmatch.go:299]: Good catch — applied in 8088377. Added json.Number to isNumericKind so a huge-integer node compares numerically against a numeric spec value instead of lexically; the toBigInt path keeps distinct large integers apart, and a regression case (equals ignoring surrounding whitespace) locks in the numeric semantics.
